sentry_sdk/tracing_utils.py

```diff
@@ -213,20 +219,14 @@ def add_query_source(span):
         is_sentry_sdk_frame = namespace is not None and namespace.startswith(
             "sentry_sdk."
         )
+        should_be_included = _module_in_list(namespace, in_app_include)
+        should_be_excluded = _is_external_source(abs_path) or _module_in_list(
+            namespace, in_app_exclude
+        )
 
-        should_be_included = not _is_external_source(abs_path)
-        if namespace is not None:
-            if in_app_exclude and _module_in_list(namespace, in_app_exclude):
-                should_be_included = False
-            if in_app_include and _module_in_list(namespace, in_app_include):
-                # in_app_include takes precedence over in_app_exclude, so doing it
-                # at the end
-                should_be_included = True
-
-        if (
-            abs_path.startswith(project_root)
-            and should_be_included
-            and not is_sentry_sdk_frame
+        if not is_sentry_sdk_frame and (
+            should_be_included
+            or (_is_in_project_root(abs_path, project_root) and not should_be_excluded)
```

Comment:
While I appreciate your efforts to help clean up our code, these refactoring changes are making it more difficult for me to review this PR, since it is harder for me to see how the logic has changed. I would appreciate if you could separate all the refactoring changes from this PR and open a separate PR for those changes – this PR should contain only the changes needed to fix the incorrect behavior.

Then, I will gladly re-review both PRs :)

Reply:
Here it is: https://github.com/getsentry/sentry-python/pull/3452.